phpBB/styles/prosilver/template/ucp_pm_viewfolder.html

```diff
@@ -1,130 +1,73 @@
-					<!-- IF S_UNREAD --><dd class="info"><!-- IF messagerow.FOLDER --><a href="{messagerow.U_FOLDER}">{messagerow.FOLDER}</a><!-- ELSE -->{L_UNKNOWN_FOLDER}<!-- ENDIF --></dd><!-- ENDIF -->
-					<dd class="mark"><input type="checkbox" name="marked_msg_id[]" value="{messagerow.MESSAGE_ID}" /></dd>
-				</dl>
-			</li>
-		<!-- END messagerow -->
-
-		</ul>
-	<!-- ELSE -->
-		<p><strong>
-			<!-- IF S_COMPOSE_PM_VIEW and S_NO_AUTH_SEND_MESSAGE -->
-				<!-- IF S_USER_NEW -->{L_USER_NEW_PERMISSION_DISALLOWED}<!-- ELSE -->{L_NO_AUTH_SEND_MESSAGE}<!-- ENDIF -->
-			<!-- ELSE -->
-				{L_NO_MESSAGES}
-			<!-- ENDIF -->
-		</strong></p>
-	<!-- ENDIF -->
-
-	<!-- IF FOLDER_CUR_MESSAGES neq 0 -->
-		<fieldset class="display-actions">
-			<div class="left-box"><label for="export_option">{L_EXPORT_FOLDER}{L_COLON} <select name="export_option" id="export_option"><option value="CSV">{L_EXPORT_AS_CSV}</option><option value="CSV_EXCEL">{L_EXPORT_AS_CSV_EXCEL}</option><option value="XML">{L_EXPORT_AS_XML}</option></select></label> <input class="button1 button button-form-bold" type="submit" name="submit_export" value="{L_GO}" /><br /></div>
-			<select name="mark_option">{S_MARK_OPTIONS}{S_MOVE_MARKED_OPTIONS}</select> <input class="button1 button button-form-bold" type="submit" name="submit_mark" value="{L_GO}" />
-			<div><a href="#" onclick="marklist('viewfolder', 'marked_msg', true); return false;">{L_MARK_ALL}</a> &bull; <a href="#" onclick="marklist('viewfolder', 'marked_msg', false); return false;">{L_UNMARK_ALL}</a></div>
-		</fieldset>
-
-		<hr />
-
-		<div class="action-bar bottom">
-			<!-- INCLUDE display_options.html -->
-			<input type="hidden" name="cur_folder_id" value="{CUR_FOLDER_ID}" />
-
-			<div class="pagination">
-				{TOTAL_MESSAGES}
-				<!-- IF .pagination -->
-					<!-- INCLUDE pagination.html -->
-				<!-- ELSE -->
-					 &bull; {PAGE_NUMBER}
-				<!-- ENDIF -->
+{% include 'ucp_header.html' %}
+
+{% if not PROMPT %}
+	{% include 'ucp_pm_message_header.html' %}
+{% endif %}
+
+{% if messagerow %}
+	<div class="pm">
+			<ul class="pm-list">
+				{% for message in messagerow %}
+					<li class="pm-item">
+						<span class="pm-primary">
+							{% if message.FOLDER_IMG_STYLE == "pm_unread" %}
+								<i class="fa fa-certificate fa-fw pm-unread" title="Unread message" aria-label="Unread message"></i>
+							{% endif %}
+
+							{% if message.PM_CLASS == "pm_marked_colour" %}
+								<i class="fa fa-star fa-fw pm-marked" title="You marked this message important!" aria-label="You marked this message important!"></i>
+							{% endif %}
+
+							<a href="{message.U_VIEW_PM}" title="{message.SUBJECT}" class="pm-title {{ message.FOLDER_IMG_STYLE == 'pm_unread' ? 'pm-unread' }}">{message.SUBJECT}</a>
+							<span class="pm-secondary">
+								<span class="pm-secondary-item pm-authour">
+									{% if message.PM_CLASS == "pm_foe_colour" %}
```

Comment:
Again, use dedicated template variables for this:
```php
'S_AUTHOR_FOE'		=> isset($foe[$row['author_id']]),
```

```twig
{% if message.S_AUTHOR_FOE %}
```
```twig

Reply:
The above expression doesn't work correctly, it changes the icon for both friends and foes.
This would work I think ($var == 'foe') ? true : false

Reply:
How is that possible? I think it must be another source of the problem.

Reply:
Yes, I tried replacing the modified query ($sql_array) with the original one and now it works fine

Reply:
Oh so I guess we need to alter some other query as well. Or maybe we need to select more data from the changed query. Anyway, I insist we display one thread only once so it's unacceptable to keep the original query. We need to solve this situation otherwise.

Reply:
Agreed displaying single thread is better imo as well. I'll go ahead and commit remaining changes for now

Reply:
I was wrong, the issue wasn't with the query :)